host/xtest/regression_8100.c

```diff
@@ -189,9 +187,7 @@ static bool check(ADBG_Case_t *c, STACK_OF(X509) *trusted,
 	X509_STORE_CTX *csc = NULL;
 	X509_VERIFY_PARAM *pm = NULL;
 	int i = 0;
-	time_t vfy_time;
-
-	memset(&vfy_time, 0, sizeof(vfy_time));
+	time_t vfy_time = { 0 };
```

Comment:
So it should be initialized with `0` instead.

Reply:
Why not using universal initializer? It makes this code agnostic of `time_t` effective type.

Reply:
`time_t` is a scalar as required by the C standard. Universal initializer is not the first choice.

Reply:
I understand that:
* `= { 0 }` is a C standard zeroies initializer for non-scalar variables.
* scalar variable mandates a scaler init value, i.e. `= 0`.
* some (maybe) opaque defined types as `time_t` need scalar treatment as scalar per standard.
* I guess there are standard types that are non-scalar per standard, as mandates `{ 0 }`.
* some types as `phtread_t` may depend on toolchain and need common scalar/non-scalar initializer. Here, look like `memset()` is a way.

Do you agree with this statement? And I to update on going P-Rs (#339, #340, https://github.com/OP-TEE/optee_client/pull/154) accordingly?

Note: An earlier comment worried about toolchain complain when using `{ 0 }` over a `struct { void *ptr; }` type. Should not worry as `{ 0 }` is a universal form of `{ NULL }`.

Reply:
Here's my take on this:

Types like `time_t` which are standardized as a scalar (and all other scalars for that matter) are initialized with `0`, unless another value makes more sense.

For optee_client we need maximum portability, so only initialize struct types (and `pthread_t`) with `memset()` unless there's a good reason not to do so.

For the rest of the gits we assume that a recent version of GCC or Clang is used so we initialize structs with `{ }` in order to avoid the more clumsy `memset()` procedure. Types like `pthread_t` which can be a scalar or a composite type are initialized with `memset()` in order to minimize the amount of future headache.

I don't think we need to use `{ 0 }` in any situation, if there is one we're probably better off with `memset()` instead (minimize future headache clause).

Let's take one PR at a time to avoid too many changes back and forth.

Reply:
@jenswi-linaro sounds good to me. Assuming we agree on this, I think this text should be added to https://github.com/OP-TEE/optee_docs/blob/master/general/coding_standards.rst.

Reply:
> Let's take one PR at a time to avoid too many changes back and forth.

ok

Reply:
PR https://github.com/OP-TEE/optee_docs/pull/19 created